sdk/cosmos/azure-cosmos/test/test_crud_subpartition.py

```diff
@@ -0,0 +1,453 @@
+                               'city': 'Redmond',
+                               'zipcode': '98052'}
+
+        created_document = created_collection.create_item(
+            body=document_definition
+        )
+
+        self.assertEqual(created_document.get('id'), document_definition.get('id'))
+        self.assertEqual(created_document.get('key'), document_definition.get('key'))
+        self.assertEqual(created_document.get('city'), document_definition.get('city'))
+        self.assertEqual(created_document.get('zipcode'), document_definition.get('zipcode'))
+
+        # read document
+        read_document = created_collection.read_item(
+            item=created_document.get('id'),
+            partition_key=[created_document.get('city'), created_document.get('zipcode')]
+        )
+
+        self.assertEqual(read_document.get('id'), created_document.get('id'))
+        self.assertEqual(read_document.get('key'), created_document.get('key'))
+        self.assertEqual(read_document.get('city'), created_document.get('city'))
+        self.assertEqual(read_document.get('zipcode'), created_document.get('zipcode'))
+
+        # Read document feed doesn't require partitionKey as it's always a cross partition query
+        documentlist = list(created_collection.read_all_items())
+        self.assertEqual(1, len(documentlist))
+
+        # replace document
+        document_definition['key'] = 'new value'
+
+        replaced_document = created_collection.replace_item(
+            item=read_document,
+            body=document_definition
+        )
+
+        self.assertEqual(replaced_document.get('key'), document_definition.get('key'))
+
+        # upsert document(create scenario)
+        document_definition['id'] = 'document2'
+        document_definition['key'] = 'value2'
+        document_definition['city'] = 'Atlanta'
+        document_definition['zipcode'] = '30363'
+
+        upserted_document = created_collection.upsert_item(body=document_definition)
+
+        self.assertEqual(upserted_document.get('id'), document_definition.get('id'))
+        self.assertEqual(upserted_document.get('key'), document_definition.get('key'))
+        self.assertEqual(upserted_document.get('city'), document_definition.get('city'))
+        self.assertEqual(upserted_document.get('zipcode'), document_definition.get('zipcode'))
+
+        documentlist = list(created_collection.read_all_items())
+        self.assertEqual(2, len(documentlist))
+
+        # delete document
+        created_collection.delete_item(item=upserted_document, partition_key=[upserted_document.get('city'),
+                                                                              upserted_document.get('zipcode')])
+
+        # query document on the partition key specified in the predicate will pass even without setting
+        # enableCrossPartitionQuery or passing in the partitionKey value
+        documentlist = list(created_collection.query_items(
```

Comment:
also we're missing tests on using incomplete partition keys for the query in the options - what happens if I run a query with only a partial partition key (aka passing in just city as the pk?)

should also add tests for passing in an incomplete partition key in the query text and checking the results are expected